### DISCUSSION TITLE
Lab_3 HW run error--Failed to get the size of the Device parameter value CL_DEVICE_NAME

### DISCUSSION THREAD

**Discussion by t881128** [bol-edu/HLS-SOC-Discussions#15] [Q&A] [answered]
When I run Lab 3, at the final step (run), I get the following message no matter I use the divice name (in run configuration/program argument) xilinx_u50_gen3x16_xdma_5_202210_1 or xilinx_u50_gen3x16_xdma_base_5. How can I fix it?
(The SW/HW emulation are success with divice name xilinx_u50_gen3x16_xdma_5_202210_1, but it didn't work out in HW case)

HOST-Error: Failed to get the size of the Device parameter value CL_DEVICE_NAME

XRT build version: 2.12.427
Build hash: 2719b6027e185000fc49783171631db03fc0ef79
Build date: 2021-10-08 22:06:50
Git branch: 2021.2
PID: 1362084
UID: 1054
[Sat Sep 17 17:25:32 2022 GMT]
HOST: HLS01
EXE: /home/CMMBCB/workspace/Lab_3/opt1/Hardware/opt1
[XRT] WARNING: dev_init failed: -1
[XRT] ERROR: Permission denied Device index 0
[XRT] ERROR: Could not open device with index '0'

**Comment by tofumanjeff** [comment]
This is due to not having the permission to access U50.
Referring to the guideline of renting FPGAs, there are two options in U50 servers, i.e., **only vitis tool** and **only u50 validation**.
The **only vitis tool** option is for compiling/building only, so you won't have the permission to access U50.
To run the HW, you have to download your compiled files, rent an **only u50 validation** server, and upload your files to the newly rented server.
Well, we know this is kind of inconvenient, but it is to prevent simultaneous access to U50 and do some load balancing.

**Comment by tofumanjeff** [comment, marked as answer]
**Update**
We found that there are some problems importing projects to a newly rented server, so we have updated the renting system.
For Lab#3, there is only one option, i.e., **vitis tool and u50 validation**. You now have complete permission to the server.
This is a temporary configuration. Upcoming labs will take longer to finish, and you may need to keep your files to continue your work.
We will update the renting system to give you a fixed account, while the password may change each time you rent a server.
Please keep track of TA's latest announcement. Thanks.
